### PR TITLE
Allow specifying glob pattern to ignore config option

### PR DESCRIPTION
**Pull request by rchaser53**
related: https://github.com/rust-lang/rustfmt/pull/3413

### CODE REVIEW

File: src/test/mod.rs [hunk 1]
```diff
@@ -560,7 +560,7 @@ fn get_config(config_file: Option<&Path>) -> Config {
         .read_to_string(&mut def_config)
         .expect("Couldn't read config");
 
-    Config::from_toml(&def_config, Path::new("tests/config/")).expect("invalid TOML")
```
Comment (scampi) [inline src/test/mod.rs]: If I understand correctly, the folder `"tests/config/"` was ignored. Where is it ignored now ?

Comment (rchaser53) [reply src/test/mod.rs]: ```tests/config/``` includes no rs files now(only toml files).
So I think ```tests/config``` does not need to be ignored.
https://github.com/rust-lang/rustfmt/tree/master/tests/config

Comment (scampi) [reply src/test/mod.rs]: indeed, thanks for checking!

File: src/ignore_path.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        })
+    }
+
+    pub fn is_match(&self, file_name: &FileName) -> bool {
+        match file_name {
+            FileName::Stdin => false,
+            FileName::Real(p) => !self
+                .ignore_set
+                .matched_path_or_any_parents(p, false)
+                .is_none(),
```
Comment (topecongiro) [inline src/ignore_path.rs]: Should use `is_ignore()` instead of `!` and `is_none()`.

File: src/config/mod.rs [hunk 3]
```diff
@@ -135,7 +135,7 @@ create_config! {
         "Report all, none or unnumbered occurrences of TODO in source file comments";
     report_fixme: ReportTactic, ReportTactic::Never, false,
         "Report all, none or unnumbered occurrences of FIXME in source file comments";
-    ignore: IgnoreList, IgnoreList::default(), false,
+    ignore: IgnoreList, IgnoreList::default(), true,
```
Comment (topecongiro) [inline src/config/mod.rs]: Would you please not make `ignore` stable in this PR? I would prefer not to mark it stable until it gets tested a bit more.

**Comment by rchaser53** [comment]
Thank you the review. I changed my code.
Should I not use rebase for review easily?

**Comment by topecongiro** [comment]
> Should I not use rebase for review easily?

You don't have to rebase, I'll just squash and merge when the PR is ready.

File: Cargo.toml [hunk 4]
```diff
@@ -57,12 +57,12 @@ bytecount = "0.5"
 unicode-width = "0.1.5"
 unicode_categories = "0.1.1"
 dirs = "1.0.4"
-globset = "0.4"
 
 # A noop dependency that changes in the Rust repository, it's a bit of a hack.
 # See the `src/tools/rustc-workspace-hack/README.md` file in `rust-lang/rust`
 # for more information.
 rustc-workspace-hack = "1.0.0"
+ignore = "0.4.6"
```
[the remark this answers is not in the document]
Comment (rchaser53) [reply Cargo.toml]: Oh Sorry.

**Comment by rchaser53** [comment]
> > Should I not use rebase for review easily?
> 
> You don't have to rebase, I'll just squash and merge when the PR is ready.

I see. I will not rebase since next time.

**Comment by euclio** [comment]
I think this may have broken how `ignore` handles workspaces? I have an ignore file at the workspace root that specifies that certain files must not be formatted within dependent crates. It works as expected with `rustfmt 1.2.0-nightly (09940a70 2019-03-27)` on my local machine but seems to have broken with a more recent rustfmt: https://travis-ci.org/Hoverbear/rust-rosetta/builds/523126071

I'm currently unable to `rustup` to the same version of `rustfmt` that my CI is using so I'm unable to test it locally and see if it's workspace-related.

**Comment by topecongiro** [comment]
@euclio Thank you for reporting the issue. I could confirm it locally and created an #3521 to track it.

**Pull request merged**